File: BroadcastReceivers/LocalBroadcastManager/app/src/main/java/com/tomerpacific/broadcastreceiver/MainActivity.java
package com.tomerpacific.broadcastreceiver;

import android.content.BroadcastReceiver;
import android.content.Intent;
import android.content.IntentFilter;
import android.support.v4.content.LocalBroadcastManager;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.view.View;

public class MainActivity extends AppCompatActivity {

    private BroadcastReceiver myBroadcastReceiver = new MyBroadcastReceiver();

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        LocalBroadcastManager.getInstance(this).registerReceiver(myBroadcastReceiver,  new IntentFilter("MY_ACTION"));
    }

    public void sendBroadcast(View view) {
        Intent intent = new Intent("MY_ACTION");
        intent.putExtra("data", "Hello World!");
        LocalBroadcastManager.getInstance(this).sendBroadcast(intent);
    }

    @Override
    protected void onStop() {
        super.onStop();
        LocalBroadcastManager.getInstance(this).unregisterReceiver(myBroadcastReceiver);
    }
}

File: BroadcastReceivers/LocalBroadcastManager/app/src/main/java/com/tomerpacific/broadcastreceiver/MainActivity.kt
package com.tomerpacific.broadcastreceiver

import androidx.appcompat.app.AppCompatActivity
import android.content.BroadcastReceiver
import android.os.Bundle
import androidx.localbroadcastmanager.content.LocalBroadcastManager
import android.content.IntentFilter
import android.content.Intent
import android.view.View

class MainActivity : AppCompatActivity() {
    private val myBroadcastReceiver: BroadcastReceiver = MyBroadcastReceiver()
    private val ACTION_KEY: String = "MY_ACTION"

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        LocalBroadcastManager.getInstance(this)
            .registerReceiver(myBroadcastReceiver, IntentFilter(ACTION_KEY))
    }

    fun sendBroadcast(view: View?) {
        val intent = Intent(ACTION_KEY)
        intent.putExtra("data", "Hello World!")
        LocalBroadcastManager.getInstance(this).sendBroadcast(intent)
    }

    override fun onStop() {
        super.onStop()
        LocalBroadcastManager.getInstance(this).unregisterReceiver(myBroadcastReceiver)
    }
}